app/src/main/java/swati4star/createpdf/util/MergeHelper.java

```diff
@@ -36,11 +43,57 @@ public MergeHelper(Activity activity, ViewFilesAdapter viewFilesAdapter) {
         mContext = mActivity;
         mViewFilesAdapter = viewFilesAdapter;
         mSharedPrefs = PreferenceManager.getDefaultSharedPreferences(mActivity);
+        mPDFEncryptUtils = new PDFEncryptionUtility(activity);
     }
 
     public void mergeFiles() {
         String[] pdfpaths = mViewFilesAdapter.getSelectedFilePath().toArray(new String[0]);
+        List<Integer> encryptedPfdPathsIndex = new ArrayList<>();
         String masterpwd = mSharedPrefs.getString(MASTER_PWD_STRING, appName);
+        int count = 0, i = 0;
+        for (String filePath : pdfpaths) {
+            if (mPDFEncryptUtils.isPDFEncrypted(filePath)) {
+                count++;
+                encryptedPfdPathsIndex.add(i);
+            }
+            i++;
+        }
+        MaterialDialog.Builder builder = new MaterialDialog.Builder(mActivity);
+        List<EditText> passwords = new ArrayList<>();
+        for (int k = 0; k < count; k++) {
```

Comment:
Here I'm trying to make edit texts for each encrypted file so that user can input the passwords.